src/main/scala/com/databricks/spark/xml/XmlInputFormat.scala

```diff
@@ -137,13 +143,23 @@ private[xml] class XmlRecordReader extends RecordReader[LongWritable, Text] {
    * @return whether it reads successfully
    */
   private def next(key: LongWritable, value: Text): Boolean = {
-    if (readUntilStartElement()) {
+    if (readUntilMatch(startTag, withinBlock = false, force = false)) {
```

Comment:
I guess here is the very core in this library, performance sensitive and affect all the functionailities .. it'd be nicer if we have tests.